test/test_rpc.py

```diff
@@ -378,6 +379,44 @@ def test_stress_light_rpc(self):
     def test_stress_heavy_rpc(self):
         self._stress_test_rpc(heavy_rpc, repeat=20, args=(torch.ones(100, 100),))
 
+    @_wrap_with_rpc
+    def test_builtin_remote_ret(self):
+        n = self.rank + 1
+        dst_rank = n % self.world_size
+        rref = dist.remote('worker{}'.format(dst_rank), torch.add,
+                           args=(torch.ones(n, n), torch.ones(n, n)))
+        self.assertEqual(rref.to_here(), torch.ones(n, n) * 2)
+
+    @_wrap_with_rpc
+    def test_multi_builtin_remote_ret(self):
+        m = 10
+        n = self.rank + 1
+        dst_rank = n % self.world_size
+        rrefs = []
+        expected = []
+        for i in range(m):
+            n = n + i
+            rrefs.append(dist.remote(
+                'worker{}'.format(dst_rank),
+                torch.add,
+                args=(torch.ones(n, n), torch.ones(n, n))
+            ))
+            expected.append(torch.ones(n, n) * 2)
+
+        def all_to_here(rrefs, values):
+            for i in range(m):
+                values.append(rrefs[i].to_here())
+
+        values = []
+        t = threading.Thread(target=all_to_here, args=(rrefs, values))
+        t.start()
+        t.join(int(get_timeout(self.id()) / 2))
```

Comment:
Why run this on a separate thread?

Reply:
I was a little worried it might take too long to finish all `to_here` calls. At a second thought, the default timeout should sufficient. I will remove this thread.